lib/orderbook/OrderBook.ts

```diff
@@ -80,18 +80,24 @@ class OrderBook {
     return this.addOwnOrder(order);
   }
 
-  public addMarketOrder = async (_order: orders.OwnOrder) => {
-    // TODO: implement
+  public addMarketOrder = async (_order: orders.MarketOrder): Promise<matchingEngine.OrderMatch[]> => {
+    const price = _order.quantity > 0 ? Number.MAX_VALUE : 0;
+    const order: orders.OwnOrder = { ..._order, price };
+
+    const result = await this.addOwnOrder(order, true);
+    delete result.remainingOrder;
```

Comment:
The remaining order is being discarded (not added to queues, not broadcasted), but I think we should return it to the client anyway.